### PR TITLE
Create an empty global modal

### PR DESCRIPTION
**Pull request by ErezBiren**
## Changes Proposed

Related to Issue #243 
I opened a Draft pull request just for code review, it's not the final PR.
I would love to get some feedback, to make sure I took the right path.

I've created an empty modal that takes as arguments:
title text, content text, and an array of actions.
each action will be translated to a button at the bottom of the modal.
each action includes a string, a callback, and a style.

@chad11111 

## UI images from Storybook

basic modal:
image

fully customized modal:
image

## Check List

- [x] The title of my pull request is a short description of the changes
- [x] This PR relates to some issue: 
- [x] I have documented the changes made (if applicable)
- [ ] I have covered the changes with unit tests
- [x] The Modal Title can be fully customized instead of just text.
- [x] The Modal Content can be fully customized instead of just text.
- [ ] Track multiple open modals.
- [x] The middle section should expand according to the content displayed up to the size of 500px after that a scroll wheel should appear.

### CODE REVIEW

**Comment by chad11111** [comment]
@ErezBiren I think u carried over unwanted changes (probably from main)

**Comment by ErezBiren** [comment]
@chad11111 
Yes, I've noticed that but wasn't sure about it.
I'll try to get rid of them and do a new PR

File: dashboard/src/components/modal/Modal.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
+// Modal.stories.ts|tsx
+
+import { ComponentStory, ComponentMeta } from "@storybook/react";
```
Comment (chad11111) [inline dashboard/src/components/modal/Modal.stories.tsx:3]: I wasn't able to get the storybook running.

image

**Review by chad11111 — CHANGES_REQUESTED** [review]

File: dashboard/src/components/modal/Modal.stories.tsx (hunk 1 above)
Comment (chad11111) [reply dashboard/src/components/modal/Modal.stories.tsx:3]: @ErezBiren other then that amazing job!

**Pull request merged**